### PR TITLE
file inclusion improvement + docs

### PR DESCRIPTION
**Pull request by Orzelius**
see commits

related to #4763

### CODE REVIEW

File: core/src/graph/actions.ts [hunk 1]
```diff
@@ -282,6 +283,18 @@ export const actionFromConfig = profileAsync(async function actionFromConfig({
 
   const dependencies = dependenciesFromActionConfig(log, config, configsByKey, definition, templateContext)
 
+  if (config.exclude?.includes("**/*")) {
```
Comment (Walther) [inline core/src/graph/actions.ts:286]: should we have equals here instead of includes, in case of something like `src/test/**/*`?

Comment (shumailxyz) [reply core/src/graph/actions.ts:286]: also, can we also add a test for this?

Comment (Orzelius) [reply core/src/graph/actions.ts:286]: `config.exclude` is an array of strings. It's checking if the array includes the exact match of "**/*"

Comment (Orzelius) [reply core/src/graph/actions.ts:286]: Writing tests was a good idea, found a bug in my original logic and removed some unwanted mutation
[e4a9dce](https://github.com/garden-io/garden/pull/4768/commits/e4a9dcef26e256511af177244f7c34b3d76c034e)

**Review by shumailxyz — APPROVED** [review]
Nice, thank you 👍

**Review by Walther — APPROVED** [review]
nice work! 🚢

**Pull request merged**